CMakeLists.txt

```diff
@@ -0,0 +1,148 @@
+# We require 3.9 because of the use of $<TARGET_OBJECTS:crt1>
+cmake_minimum_required(VERSION 3.9)
+project(MUSL)
+enable_language(C ASM)
+include(CheckCCompilerFlag)
+
+if(WASM)
+  set(ARCH "wasm32")
+elseif(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "x86_64")
+  set(ARCH "x86_64")
+else()
+  error("Could not detect ARCH")
+endif(WASM)
+
+set(EMPTY_LIB_NAMES m rt pthread crypt util xnet resolv dl)
+
+include_directories(
+  ${MUSL_SOURCE_DIR}/arch/${ARCH}
+  ${MUSL_SOURCE_DIR}/arch/generic
+  ${MUSL_BINARY_DIR}/src/internal
+  ${MUSL_SOURCE_DIR}/src/internal
+  ${MUSL_BINARY_DIR}/include
+  ${MUSL_SOURCE_DIR}/include
+)
+
+add_definitions(
+ -D_XOPEN_SOURCE=700
+ -std=c99
+ -ffreestanding
+ -nostdinc
+)
+
+set(OPTIONAL_FLAGS
+ -fomit-frame-pointer
+ -fno-unwind-tables
+ -fno-asynchronous-unwind-tables
+ -ffunction-sections
+ -fdata-sections
+ -fexcess-precision=standard
+ -frounding-math
+ -Wa,--noexecstack
+)
+
+set(WARNING_FLAGS
+ -Werror
+ -Wno-empty-body
+ -Wno-incompatible-library-redeclaration
+ -Wno-shift-op-parentheses
+ -Wno-tautological-unsigned-zero-compare
+ -Wno-tautological-constant-out-of-range-compare
+ -Wno-tautological-unsigned-enum-zero-compare
+ -Wno-ignored-attributes
+ -Wno-format
+ -Wno-bitwise-op-parentheses
+ -Wno-logical-op-paretheses
+ -Wno-string-plus-int
+ -Wno-pointer-sign
+ -Wno-dangling-else
+ -Wno-absolute-value
+ -Wno-parentheses
+ -Wno-unknown-pragmas
+)
+
+foreach(flag IN LISTS OPTIONAL_FLAGS WARNING_FLAGS)
+  check_c_compiler_flag(${flag} FLAG_${flag})
+  if(FLAG_${flag})
+    add_definitions(${flag})
+  endif(FLAG_${flag})
+endforeach(flag)
+
+set(CMAKE_SHARED_LINKER_FLAGS "-nostdlib -Wl,-e,_dlstart")
+
+add_custom_command(
+  OUTPUT ${MUSL_BINARY_DIR}/src/internal/version.h
+  COMMAND
+  echo '\#define VERSION \"1\"' > ${MUSL_BINARY_DIR}/src/internal/version.h
+  DEPENDS
+  ${MUSL_SOURCE_DIR}/VERSION
+)
+
+add_custom_command(
+  OUTPUT ${MUSL_BINARY_DIR}/include/bits/syscall.h
+  COMMAND
+  cp ${MUSL_SOURCE_DIR}/arch/${ARCH}/bits/syscall.h.in ${MUSL_BINARY_DIR}/include/bits/syscall.h &&
+  sed -n -e s/__NR_/SYS_/p < ${MUSL_SOURCE_DIR}/arch/${ARCH}/bits/syscall.h.in >> ${MUSL_BINARY_DIR}/include/bits/syscall.h
```

Comment:
Any chance we could eventually remove the use of sed here (and do it in, say, python?). It would make the windows build much simpler. (not necessarily for this PR, just an observation).

Reply:
Btw I assume you haven't actually tested this on Windows?

Reply:
Errrr... 


(how do you normally go about doing that?)

Reply:
fire up the windows box and try it out :D 
I'm actually about to do that anyway, last time I was sheriff I started an updated of the toolchain cmake file to fix compiler-rt and make it generally better, so maybe i'll give this one a shot too.